Typeguard fails with 8.7.0

It worked fine with 8.6.1
See https://github.com/hacf-fr/renault-api/pull/1234

```
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
.nox/typeguard-3-12/lib/python3.12/site-packages/typeguard/_importhook.py:98: in exec_module
    super().exec_module(module)
.nox/typeguard-3-12/lib/python3.12/site-packages/renault_api/kamereon/models.py:15: in <module>
    from . import helpers
<frozen importlib._bootstrap>:1360: in _find_and_load
    ???
<frozen importlib._bootstrap>:1331: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:935: in _load_unlocked
    ???
.nox/typeguard-3-12/lib/python3.12/site-packages/typeguard/_importhook.py:98: in exec_module
    super().exec_module(module)
<frozen importlib._bootstrap_external>:991: in exec_module
    ???
<frozen importlib._bootstrap_external>:1129: in get_code
    ???
.nox/typeguard-3-12/lib/python3.12/site-packages/typeguard/_importhook.py:87: in source_to_code
    return _call_with_frames_removed(
.nox/typeguard-3-12/lib/python3.12/site-packages/typeguard/_importhook.py:47: in _call_with_frames_removed
    return f(*args, **kwargs)
E     File "/home/runner/work/renault-api/renault-api/.nox/typeguard-3-12/lib/python3.12/site-packages/renault_api/kamereon/helpers.py", line 3
E       from __future__ import annotations
E       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   SyntaxError: from __future__ imports must occur at the beginning of the file
```

--- a/setup.py
+++ b/setup.py
@@ -49,7 +49,7 @@ setup(
     install_requires=[
         "marshmallow>=3.18.0,",
         "typing-inspect>=0.9.0",
-        "typeguard>=4.0.0",
+        "typeguard>=4.0,<5",
         # Need `dataclass_transform(field_specifiers)`
         "typing-extensions>=4.2.0; python_version<'3.11'",
     ],

--- a/setup.py
+++ b/setup.py
@@ -48,8 +48,8 @@ setup(
     python_requires=">=3.8",
     install_requires=[
         "marshmallow>=3.18.0,",
-        "typing-inspect~=0.9.0",
-        "typeguard~=4.0.0",
+        "typing-inspect>=0.9.0",
+        "typeguard>=4.0.0",
         # Need `dataclass_transform(field_specifiers)`
         "typing-extensions>=4.2.0; python_version<'3.11'",
     ],

